src/shogun/mathematics/linalg/linop/SparseMatrixOperator.cpp

```diff
@@ -129,8 +129,8 @@ SparsityStructure* CSparseMatrixOperator<T>::get_sparsity_structure(
 		}
 
 		// create the sparsity structure using the final power
-		int32_t* outerIndexPtr=const_cast<int32_t*>(matrix_power.outerIndexPtr());
-		int32_t* innerIndexPtr=const_cast<int32_t*>(matrix_power.innerIndexPtr());
+		index_t* outerIndexPtr=const_cast<index_t*>(matrix_power.outerIndexPtr());
+		index_t* innerIndexPtr=const_cast<index_t*>(matrix_power.innerIndexPtr());
```

Comment:
Hi @vigsterkr , so the change in here is actually wrong, since `const_cast from 'StorageIndex *' (aka 'int *') to 'index_t *' (aka 'long long *') is not allowed`. But as you know the StorageIndex* is actually more complex than int32_t which can't change directly. So what should I do next? Thank you!